plugins/git-release-manager/src/hooks/useResponseSteps.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState } from 'react';
+
+import { ResponseStep } from '../types/types';
+
+const RESPONSE_STEP_FAILURE_ABORT: ResponseStep = {
+  message: 'Skipped due to error in previous step',
+  icon: 'failure',
+};
+
+export function useResponseSteps() {
+  const [responseSteps, setResponseSteps] = useState<ResponseStep[]>([]);
+
+  const addStepToResponseSteps = (responseStep: ResponseStep) => {
```

Comment:
in general i find it a nice thing for consumers of a hook if this type of returned function is wrapped in useCallback with no dependencies, so one can rely on it never changing under any circumstances (because they may also have it in their dependency lists)

Reply:
`responseSteps` has to be a dependency though 🤔